File: 프로그래머스/0/120885. 이진수 더하기/이진수 더하기.js
function solution(bin1, bin2) { 
    let num = +bin1 + +bin2;
    let arr = String(num).split('').map(Number);
    for(let i = arr.length - 1; i > 0; i--){ // 2번째 숫자부터 끝까지
        if(arr[i] === 2){
            arr[i] = 0;
            arr[i-1] += 1;
        }
        else if(arr[i] === 3){
            arr[i] = 1;
            arr[i-1] += 1;
        }
    }
    
    if(arr[0] === 2){
        arr[0] = 0;
        return [1].concat(arr).join('');
    }
    else if(arr[0] === 3){
        arr[0] = 1;
        return [1].concat(arr).join('');
    }
    else{
        return arr.join('')
    }
    
    
    
}

File: 프로그래머스/0/120885. 이진수 더하기/이진수 더하기.py
def solution(bin1, bin2):
    # bin1 = 1001
    # bin2 = 1111
    
    answer = ""
    plusValue = 0
    minLen = 0
    rBin1 = bin1[::-1] # 1001
    rBin2 = bin2[::-1] # 1111
    minBin = ""

    if len(rBin1)>len(rBin2):
        minLen = len(rBin1) - len(rBin2)
        minBin = "rBin2"
    else:
        minLen = len(rBin2) - len(rBin1)
        minBin = "rBin1"

    for i in range(minLen):
        if minBin == "rBin1":
            rBin1 += "0"
        else:
            rBin2 += "0"

    print(rBin1)
    for i in range(len(rBin1)):
        num1 = int(rBin1[i])
        num2 = int(rBin2[i])
        value = num1+num2+plusValue

        if value == 0:
            plusValue = 0
            answer += "0"
        elif value == 1:
            plusValue = 0
            answer += "1"
        elif value == 2:
            plusValue = 1
            answer += "0"
        elif value == 3:
            plusValue = 1
            answer += "1"
    if plusValue == 1:
        answer += "1"

    return answer[::-1]